docs/architecture/adr-038-add-support-for-wasm-light-clients.md

```diff
@@ -0,0 +1,77 @@
+# ADR 38: Add support for wasm based light client
+## Changelog
+- {date}: {changelog}
+## Status
+{DRAFT | PROPOSED} Not Implemented
+> Please have a look at the [PROCESS](./PROCESS.md#adr-status) page.
+> Use DRAFT if the ADR is in a draft stage (draft PR) or PROPOSED if it's in review.
+## Abstract
+> "If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the ADR.
+> A short (~200 word) description of the issue being addressed.
+
+Currently in the SDK, light clients are part of cosmos sdk codebase. What this means that, anytime a blockchain built upon cosmos-sdk need to add
+support for new light client, cosmos-sdk need to be modified and all validator nodes of that blockchain need to be updated with custom version of
+cosmos-sdk. 
+
+To remedy these shortcomings, we are proposing a WASM VM to host light client bytecode, which allows easier upgrading of existing light clients
+as well as adding support for new light clients.
+## Context
+> This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply describing facts. It should clearly explain the problem and motivation that the proposal aims to resolve.
+> {context body}
+>
+Currently in the SDK, light clients are defined as part of the codebase and lives at ``. To add support for new light client or
+update existing light client in the event of security issue or consensus update, we need to modify the codebase to add another light client
+module at the location specified and integrate it as light client at other places in the codebase. Apart from this, individual blockchains built upon cosmos-sdk
+need to update all their validator nodes to latest version to add support for this light client. This entire process is tedious and time consuming.
+In case a blockchain wants to add support for niche light client, it need to fork cosmos-sdk and modify codebase to add support for it. This creates overhead of
+maintaining that fork against mainstream cosmos-sdk release.
+
+As the sdk is used by more and more blockchain, either every blockchain need to maintain fork of cosmos-sdk to add support for niche light client and
+cosmos-sdk need to be frequently updated to add support for major light clients.
+
+We are proposing simplifying this work-flow by integrating a WASM light client module which makes adding support for new light client a simple transaction.
+It underneath uses WASM VM to run light client bytecode written in rust. Adding support for new light client requires writing light client in wasm compilable rust
```

Comment:
These sentences don't flow well; try something like:

"The light client bytecode, written in Wasm-compilable Rust runs inside a Wasmer VM. The Wasm light client submodule exposes a proxy light client interface that routes incoming messages to the appropriate handler function, inside the Wasm VM for execution."